### PR TITLE
Unit tests for Airlock API

### LINKED ISSUE #2016: Airlock tests and schema validation
## Description

The second part of the implementation of the api stubs PR.

### PR DESCRIPTION
**Pull request by anatbal**
# PR fixes #2016 

## What is being addressed
- Adding unit tests for all new airlock api endpoints
- Remove todos that were related to testing

### CODE REVIEW

File: api_app/tests_ma/test_db/test_repositories/test_airlock_request_repository.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+def test_update_airlock_request_status_updates_airlock_request_with_the_right_status(airlock_request_mock, airlock_request_repo):
+    airlock_request_item_to_create = airlock_request_mock
+    new_status = AirlockRequestStatus.Submitted
+    user = create_test_user()
+    airlock_request = airlock_request_repo.update_airlock_request_status(airlock_request_item_to_create, new_status, user)
+
+    assert airlock_request.status == new_status
+
+
+def test_update_airlock_request_status_fails_on_validation_wrong_status(airlock_request_mock, airlock_request_repo):
```
Comment (SharonHart) [inline api_app/tests_ma/test_db/test_repositories/test_airlock_request_repository.py:74]: Try to see if it's easy to add tests for all of the conditions in _validate_status_update
Maybe with `@parameterized`

File: api_app/tests_ma/test_api/test_routes/test_airlock_resource_helpers.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        resourceType=AirlockResourceType.AirlockReview,
+        workspaceId=WORKSPACE_ID,
+        requestId=AIRLOCK_REQUEST_ID,
+        reviewDecision=AirlockReviewDecision.Approved,
+        decisionExplanation="test explaination"
+    )
+    return airlock_review
+
+
+class TestAirlockResourceHelpers:
```
Comment (SharonHart) [inline api_app/tests_ma/test_api/test_routes/test_airlock_resource_helpers.py:101]: I think that defining it under a class is redundant?

File: api_app/tests_ma/test_api/test_routes/test_airlock.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        id=airlock_request_id,
+        workspaceId=workspace_id,
+        businessJustification="test business justification",
+        requestType="import",
+        status=AirlockRequestStatus.InReview
+    )
+    return updated_airlock_request
+
+
+def sample_airlock_review_object(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
```
Comment (SharonHart) [inline api_app/tests_ma/test_api/test_routes/test_airlock.py:94]: method arguments are unused

File: api_app/tests_ma/test_api/test_routes/test_airlock.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+@pytest.fixture
+def sample_airlock_review_input_data():
+    return {
+        "reviewDecision": "approved",
+        "decisionExplanation": "the reason why this request was approved/rejected"
+    }
+
+
+def sample_status_changed_event(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
```
Comment (SharonHart) [inline api_app/tests_ma/test_api/test_routes/test_airlock.py:36]: unused?

File: api_app/tests_ma/test_api/test_routes/test_airlock.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        id=airlock_request_id,
+        workspaceId=workspace_id,
+        businessJustification="test business justification",
+        requestType="import",
+        status=AirlockRequestStatus.Submitted
+    )
+    return updated_airlock_request
+
+
+def sample_approved_airlock_request_object(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
```
Comment (SharonHart) [inline api_app/tests_ma/test_api/test_routes/test_airlock.py:72]: if they just differ by status, make it an argument for the tests to pass

**Review by SharonHart — CHANGES_REQUESTED** [review]
Minor comments

File: api_app/tests_ma/test_db/test_repositories/test_airlock_request_repository.py (hunk 1 above)
Comment (anatbal) [reply api_app/tests_ma/test_db/test_repositories/test_airlock_request_repository.py]: Done

File: api_app/tests_ma/test_api/test_routes/test_airlock_resource_helpers.py (hunk 2 above)
Comment (anatbal) [reply api_app/tests_ma/test_api/test_routes/test_airlock_resource_helpers.py]: Done

File: api_app/tests_ma/test_api/test_routes/test_airlock.py (hunk 3 above)
Comment (anatbal) [reply api_app/tests_ma/test_api/test_routes/test_airlock.py]: Done

File: api_app/tests_ma/test_api/test_routes/test_airlock.py (hunk 4 above)
Comment (anatbal) [reply api_app/tests_ma/test_api/test_routes/test_airlock.py]: Yes, removed

File: api_app/tests_ma/test_api/test_routes/test_airlock.py (hunk 5 above)
Comment (anatbal) [reply api_app/tests_ma/test_api/test_routes/test_airlock.py]: Done

File: api_app/tests_ma/test_db/test_repositories/test_airlock_review_repository.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    airlock_review_input_mock.decisionExplanation = "test explnation"
+    return airlock_review_input_mock
+
+
+def test_create_airlock_review_item_with_the_right_values(sample_airlock_review_input, airlock_review_repo):
+    airlock__review = airlock_review_repo.create_airlock_review_item(sample_airlock_review_input, WORKSPACE_ID, AIRLOCK_REQUEST_ID)
+
+    assert airlock__review.resourceType == AirlockResourceType.AirlockReview
+    assert airlock__review.workspaceId == WORKSPACE_ID
+    assert airlock__review.requestId == AIRLOCK_REQUEST_ID
```
[the remark this answers is not in the document]
Comment (anatbal) [reply api_app/tests_ma/test_db/test_repositories/test_airlock_review_repository.py:29]: Done

**Review by SharonHart — APPROVED** [review]

**Pull request merged**